### PR TITLE
Add -print-zero-stats to fix ensure_no_astgen.swift in release mode

### PR DESCRIPTION
**Pull request by rjmansfield**
as stats collection depends on asserts enabled or forced stats collection.

Fix diagnostic when compiler is built without statistics support.

### CODE REVIEW

File: test/ScanDependencies/ensure_no_astgen.swift [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // REQUIRES: swift_swift_parser
+// REQUIRES: asserts
```
Comment (hamishknight) [inline test/ScanDependencies/ensure_no_astgen.swift:2]: We ought to be able to run this test in both debug and non-debug builds. I assume we're seeing differing behavior here due to 5025964c2eb8d2a4622f619a6a5595aff45a4943, which only changed the release (with `LLVM_ENABLE_STATS` disabled) behavior to not print zero counters. We probably ought to add a flag to control whether zero counters are printed in both debug and release builds.

Comment (rjmansfield) [reply test/ScanDependencies/ensure_no_astgen.swift]: Ok, how about `-print-zero-stats` then ?

File: lib/Basic/Statistic.cpp [hunk 2]
```diff
@@ -458,7 +462,7 @@ UnifiedStatsReporter::printAlwaysOnStatsAndTimers(raw_ostream &OS) {
     auto &C = getFrontendCounters();
 #define FRONTEND_STATISTIC(TY, NAME)                          \
     do {                                                      \
-      if (C.NAME) {                                           \
+      if (C.NAME || IsPrintingZeroStats) {                    \
```
Comment (hamishknight) [inline lib/Basic/Statistic.cpp:465]: You'll also want to check for this in `publishAlwaysOnStatsToLLVM` for the `Stat = 0` assignment to ensure consistent behaviour across debug and release builds

**Review by hamishknight — COMMENTED** [review]
Thanks! LGTM modulo one comment

**Review by hamishknight — APPROVED** [review]

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by rjmansfield** [comment]
```
Test Case '-[SwiftDocCTests.DeclarationsRenderSectionTests testAlternateDeclarations]' started.
/Users/ec2-user/jenkins/workspace/swift-PR-macos/branch-main/swift-docc/Tests/SwiftDocCTests/Rendering/DeclarationsRenderSectionTests.swift:147: error: -[SwiftDocCTests.DeclarationsRenderSectionTests testAlternateDeclarations] : XCTAssertEqual failed: ("1") is not equal to ("2")
Test Case '-[SwiftDocCTests.DeclarationsRenderSectionTests testAlternateDeclarations]' failed (1.262 seconds).

```
swift-docc test failure appears unrelated as it is happening during other CI runs.

**Comment by hamishknight** [comment]
@swift-ci please smoke test

**Pull request merged**